### PR TITLE
CloudWatch Logs: Skip caching for log queries

### PR DESCRIPTION
**Pull request by aocenas**
Because of how CloudWatch logs queries work they are not cacheable by default and caching resulted in errors due to too many retries.

This adds x-skip-cache header to log queries which means caching is always skipped.

### CODE REVIEW

File: docs/sources/enterprise/query-caching.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ You can make a panel retrieve fresh data more frequently by increasing the **Max
 
 Query caching works for all [Enterprise data sources](https://grafana.com/grafana/plugins/?type=datasource&enterprise=1), and it works for the following [built-in data sources]({{< relref "../datasources/_index.md" >}}):
 
-- CloudWatch
+- CloudWatch Metrics
```
Comment (aocenas) [inline docs/sources/enterprise/query-caching.md:34]: @sunker does it make sense to name it like this? Or do we have some other way to differentiate logs vs metrics for cloud watch?

**Review by kminehart — APPROVED** [review]

**Comment by kminehart** [comment]
FYI this header is defined here:

https://github.com/grafana/grafana-enterprise/blob/96ca7804b4f014d488d902d10f51d80123656571/src/pkg/extensions/caching/middleware/query_middleware.go#L33

File: docs/sources/enterprise/query-caching.md (hunk 1 above)
Comment (sunker) [reply docs/sources/enterprise/query-caching.md:34]: No I think that looks good

**Review by sunker — APPROVED** [review]
LGTM

**Pull request merged**